services/ux-backend/handlers/onboarding_tokens/handler.go

```diff
@@ -15,64 +15,63 @@ import (
 	"time"
 
 	"github.com/google/uuid"
-	"github.com/red-hat-storage/ocs-operator/v4/services/types"
+	"github.com/red-hat-storage/ocs-operator/v4/services"
 	"k8s.io/klog/v2"
 )
 
-const onboardingPrivateKeyFilePath = "/etc/private-key/key"
-
-func OnboardingTokensHandler(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
+const (
+	onboardingPrivateKeyFilePath = "/etc/private-key/key"
+	ContentTypeTextPlain         = "text/plain"
+)
 
-	var err error
+func HandleMessage(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
 	switch r.Method {
 	case "POST":
+		handlePost(w, r, tokenLifetimeInHours)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
 
-		onboardingToken, err := generateOnboardingToken(tokenLifetimeInHours)
-		if err != nil {
-			klog.Errorf("failed to get onboardig token: %v", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Header().Set("Content-Type", "text/text")
-			_, err = w.Write([]byte("Failed to generate token"))
-
-			if err != nil {
-				klog.Errorf("failed write data to response writer, %v", err)
-			}
-			return
-		}
+func handlePost(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
+	if onboardingToken, err := generateOnboardingToken(tokenLifetimeInHours); err != nil {
+		klog.Errorf("failed to get onboardig token: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Header().Set("Content-Type", ContentTypeTextPlain)
 
+		if _, err := w.Write([]byte("Failed to generate token")); err != nil {
+			klog.Errorf("failed write data to response writer, %v", err)
+		}
+	} else {
 		klog.Info("onboarding token generated successfully")
 		w.WriteHeader(http.StatusOK)
-		w.Header().Set("Content-Type", "text/text")
+		w.Header().Set("Content-Type", ContentTypeTextPlain)
 
-		_, err = w.Write([]byte(onboardingToken))
-		if err != nil {
+		if _, err = w.Write([]byte(onboardingToken)); err != nil {
 			klog.Errorf("failed write data to response writer: %v", err)
-			return
 		}
+	}
+}
 
-	default:
-		klog.Info("Only POST method should be used to send data to this endpoint /onboarding-tokens")
-		w.WriteHeader(http.StatusMethodNotAllowed)
-		w.Header().Set("Content-Type", "text/text")
-		_, err = w.Write([]byte(fmt.Sprintf("Unsupported method : %s", r.Method)))
-		if err != nil {
-			klog.Errorf("failed write data to response writer: %v", err)
-		}
-		return
+func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
+	klog.Info("Only POST method should be used to send data to this endpoint /onboarding-tokens")
+	w.WriteHeader(http.StatusMethodNotAllowed)
+	w.Header().Set("Content-Type", ContentTypeTextPlain)
+
```

Comment:
minor: MDN [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) says `server must generate an Allow header field in a 405 status code response`